Sources/FluentPostgresDriver/FluentPostgresDatabase.swift

```diff
@@ -42,6 +42,20 @@ extension _FluentPostgresDatabase: Database {
             closure(_FluentPostgresDatabase(database: $0, context: self.context))
         }
     }
+
+    func transaction<T>(_ transaction: @escaping (Database) throws -> EventLoopFuture<T>) -> EventLoopFuture<T> {
+        return self.simpleQuery("BEGIN TRANSACTION").flatMap { results in
+            do {
+                return try transaction(self).flatMap { res in
+                    return self.simpleQuery("END TRANSACTION").transform(to: res)
```

Comment:
At this point we are assuming that the operation in the transaction always works. You need to chain a catch call (I forgot the exact name of the method) onto the `transaction().flatMap` call, so if the operation in the transaction fails, we also rollback for that.

```swift
return try transaction(self).flatMap { res in
    return self.simpleQuery("END TRANSACTION").transform(to: res)
}.catchFlatMap {
    return self.database.simpleQuery("ROLLBACK")
}
```

Or something like that.